src/vhost_user/slave_req_handler.rs

```diff
@@ -690,24 +682,23 @@ impl<S: VhostUserSlaveReqHandler> SlaveReqHandler<S> {
     fn check_attached_rfds(
         &self,
         hdr: &VhostUserMsgHeader<MasterReq>,
-        rfds: Option<Vec<RawFd>>,
-    ) -> Result<Option<Vec<RawFd>>> {
+        rfds: &Option<Vec<RawFd>>,
+    ) -> Result<()> {
```

Comment:
The change may cause file descriptor leakage , such as call site at line 291. Functionality of "close_rfds()" should be reserved.

Reply:
Manually closing fds is a pretty strong code smell.  We shouldn't be relying on code reviews to catch leaks like this.

It seems to me like the `recv_header()` method should be returning `File`s rather than `RawFd`s.

Reply:
I replaced `RawFd`s in connection.rs with `File`s as suggested. As a result, I needed to update both the master handler and the slave handler.
While I used `File` at places where we had transferred ownership of `RawFd`, I used `&dyn RawFd` instead at places where ownership wasn't transferred.